Issue27

Michedev opened the pull request:
Close #27 
Implement SchoolController.addToDB(Student) method and Database.add(Student)

Michedev committed:
Implement base test of SchoolController.add(Student) and generated
method implementation

Michedev committed:
Add factory method for students in SchoolControllerTest

Michedev committed:
Add control that add to database if only student id is not present.
Add negative test for addToDB with stub method.

Michedev committed:
Fixed error on SchoolControllerTest and SchoolController

Michedev committed:
Solved conficts with master

LorenzoBettini commented on com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:
```diff
@@ -48,15 +48,39 @@ public void testGetStudentByIdWithBadIndex() {
+		when(database.exists(anyString())).thenReturn(false);
+		assertTrue(schoolController.addToDB(student));
+		verify(database, times(1)).add(student);
+
```
Please remove this empty line

LorenzoBettini commented on com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:
```diff
@@ -48,15 +48,39 @@ public void testGetStudentByIdWithBadIndex() {
+		when(database.exists(anyString())).thenReturn(true);
+		assertFalse(schoolController.addToDB(student));
+		verify(database, times(1)).exists(student.getId());
+
```
Please remove this empty line

LorenzoBettini commented on com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:
```diff
@@ -48,15 +48,39 @@ public void testGetStudentByIdWithBadIndex() {
+		Student student = factoryStudent("1", "Pippo");
+		when(database.exists(anyString())).thenReturn(true);
+		assertFalse(schoolController.addToDB(student));
+		verify(database, times(1)).exists(student.getId());
```
You might also want to check that database.add is NOT called

LorenzoBettini commented:
You also need to merge with master, since in the meantime another PR was merged and your PR cannot be merged: fetch from master, merge your branch with master and resolve conflicts, push your branch again (after you addressed all my other comments)

Michedev committed:
Merge remote-tracking branch 'origin/master' into Issue27

Conflicts:
	com.examples.exercise/src/main/java/com/examples/exercise/Database.java

Michedev committed:
Update methods and DbMongo implementation.

Insert verify on 'add' method in junit tests.
Insert empty implementation of 'add' method in DbMongo class.

Michedev commented:
Sorry i read now about the conflicts and now i solved them

Michedev commented:
Test coverage is decreased because now DbMongo implemenets also the "add" method created by me for the issue
